torch/testing/_internal/dist_utils.py

```diff
@@ -56,6 +64,9 @@ def new_test_method(self, *arg, **kwargs):
 
         self.worker_id = self.rank
 
+        if faulty_messages:
```

Comment:
I still didn't follow how the trap works here. Looks like the trap sets `TEST_CONFIG.build_rpc_backend_options`, which is only invoked below in `self.rpc_backend_options`. How does the trap modified the value of `faulty_messages` before this line?

Reply:
Linking the reply re setting the trap above: https://github.com/pytorch/pytorch/pull/33636#discussion_r393434900